README.TXT
TCPDF - README
============================================================

I WISH TO IMPROVE AND EXPAND TCPDF BUT I NEED YOUR SUPPORT.
PLEASE MAKE A DONATION:
http://sourceforge.net/donate/index.php?group_id=128076

------------------------------------------------------------

Name: TCPDF
Version: 6.0.099
Release date: 2014-11-15
Author:	Nicola Asuni

Copyright (c) 2002-2014:
	Nicola Asuni
	Tecnick.com LTD
	www.tecnick.com

URLs:
	http://www.tcpdf.org
	http://www.sourceforge.net/projects/tcpdf

Description:
	TCPDF is a PHP class for generating PDF files on-the-fly without requiring external extensions.
	This library includes also a class to extract data from existing PDF documents and
	classes to generate 1D and 2D barcodes in various formats.

Main Features:
    * no external libraries are required for the basic functions;
    * all standard page formats, custom page formats, custom margins and units of measure;
    * UTF-8 Unicode and Right-To-Left languages;
    * TrueTypeUnicode, OpenTypeUnicode v1, TrueType, OpenType v1, Type1 and CID-0 fonts;
    * font subsetting;
    * methods to publish some XHTML + CSS code, Javascript and Forms;
    * images, graphic (geometric figures) and transformation methods;
    * supports JPEG, PNG and SVG images natively, all images supported by GD (GD, GD2, GD2PART, GIF, JPEG, PNG, BMP, XBM, XPM) and all images supported via ImagMagick (http:  www.imagemagick.org/www/formats.html)
    * 1D and 2D barcodes: CODE 39, ANSI MH10.8M-1983, USD-3, 3 of 9, CODE 93, USS-93, Standard 2 of 5, Interleaved 2 of 5, CODE 128 A/B/C, 2 and 5 Digits UPC-Based Extension, EAN 8, EAN 13, UPC-A, UPC-E, MSI, POSTNET, PLANET, RMS4CC (Royal Mail 4-state Customer Code), CBC (Customer Bar Code), KIX (Klant index - Customer index), Intelligent Mail Barcode, Onecode, USPS-B-3200, CODABAR, CODE 11, PHARMACODE, PHARMACODE TWO-TRACKS, Datamatrix, QR-Code, PDF417;
    * JPEG and PNG ICC profiles, Grayscale, RGB, CMYK, Spot Colors and Transparencies;
    * automatic page header and footer management;
    * document encryption up to 256 bit and digital signature certifications;
    * transactions to UNDO commands;
    * PDF annotations, including links, text and file attachments;
    * text rendering modes (fill, stroke and clipping);
    * multiple columns mode;
    * no-write page regions;
    * bookmarks, named destinations and table of content;
    * text hyphenation;
    * text stretching and spacing (tracking);
    * automatic page break, line break and text alignments including justification;
    * automatic page numbering and page groups;
    * move and delete pages;
    * page compression (requires php-zlib extension);
    * XOBject Templates;
	* Layers and object visibility.
	* PDF/A-1b support.

Installation (full instructions on http:  www.tcpdf.org):
	1. copy the folder on your Web server
	2. set your installation path and other parameters on the config/tcpdf_config.php
	3. call the examples/example_001.php page with your browser to see an example

Source Code Documentation:
	http://www.tcpdf.org

Additional Documentation:
	http://www.tcpdf.org

License:
	Copyright (C) 2002-2014  Nicola Asuni - Tecnick.com LTD

	TCPDF is free software: you can redistribute it and/or modify it
	under the terms of the GNU Lesser General Public License as
	published by the Free Software Foundation, either version 3 of the
	License, or (at your option) any later version.

	TCPDF is distributed in the hope that it will be useful, but
	WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
	See the GNU Lesser General Public License for more details.

	You should have received a copy of the License
	along with TCPDF. If not, see
	<http://www.tecnick.com/pagefiles/tcpdf/LICENSE.TXT>.

	See LICENSE.TXT file for more information.

Third party fonts:

	This library may include third party font files released with different licenses.
	
	All the PHP files on the fonts directory are subject to the general TCPDF license (GNU-LGPLv3),
	they do not contain any binary data but just a description of the general properties of a particular font.
	These files can be also generated on the fly using the font utilities and TCPDF methods.
	
	All the original binary TTF font files have been renamed for compatibility with TCPDF and compressed using the gzcompress PHP function that uses the ZLIB data format (.z files).
	
	The binary files (.z) that begins with the prefix "free" have been extracted from the GNU FreeFont collection (GNU-GPLv3).
	The binary files (.z) that begins with the prefix "pdfa" have been derived from the GNU FreeFont, so they are subject to the same license.
	For the details of Copyright, License and other information, please check the files inside the directory fonts/freefont-20120503
	Link : http://www.gnu.org/software/freefont/
	
	The binary files (.z) that begins with the prefix "dejavu" have been extracted from the DejaVu fonts 2.33 (Bitstream) collection.
	For the details of Copyright, License and other information, please check the files inside the directory fonts/dejavu-fonts-ttf-2.33
	Link : http://dejavu-fonts.org
	
	The binary files (.z) that begins with the prefix "ae" have been extracted from the Arabeyes.org collection (GNU-GPLv2).
	Link : http://projects.arabeyes.org/

ICC profile:
    TCPDF includes the sRGB.icc profile from the icc-profiles-free Debian package:
    https://packages.debian.org/source/stable/icc-profiles-free


============================================================


CHANGELOG.TXT
6.0.099 (2014-11-15)
	- Added basic support for nested SVG images (adapted PR from SamMousa).
	- A bug related to setGDImageTransparency() was fixed (thanks to Maarten Boerema).

6.0.098 (2014-11-08)
	- Bug item #996 "getCharBBox($char) returns incorrect results for TTF glyphs without outlines" was fixed.
	- Bug item #991 "Text problem with SVG" was fixed (only the font style part).

6.0.097 (2014-10-20)
	- Bug item #988 "hyphenateText - charmin parameter not work" was fixed.
	- New 1D barcode method to print pre-formatted IMB - Intelligent Mail Barcode - Onecode - USPS-B-3200.

6.0.096 (2014-10-06)
	- Bug item #982 "Display style is not inherited in SVG" was fixed.
	- Bug item #984 "Double quote url in CSS" was fixed.

6.0.095 (2014-10-02)
	- Bug item #979 "New Timezone option overwriting current timezone" was fixed.

6.0.094 (2014-09-30)
	- Bug item #978 "Variable Undefined: $cborder" was fixed.

6.0.093 (2014-09-02)
	- Security fix: some serialize/unserialize methods were replaced with json_encode/json_decode to avoid a potential object injection with user supplied content. Thanks to ownCloud Inc. for reporting this issue.
	- K_TIMEZONE constant was added to the default configuration to suppress date-time warnings.

6.0.092 (2014-09-01)
	- Bug item #956 "Monospaced fonts are not alignd at the baseline" was fixed.
	- Bug item #964 "Problem when changing font size" was fixed.
	- Bug item #969 "ImageSVG with radialGradient problem" was fixed.
	- sRGB.icc file was replaced with the one from the Debian package icc-profiles-free (2.0.1+dfsg-1) 

6.0.091 (2014-08-13)
	- Issue #325"Division by zero when css fontsize equals 0" was fixed.

6.0.090 (2014-08-08)
	- Starting from this version TCPDF is also available in GitHub at https://github.com/tecnickcom/TCPDF
	- Function getmypid() was removed for better compatibility with shared hosting environments.
	- Support for pulling SVG stroke opacity value from RGBa color was mergeg [adf006].
	- Bug item #951 "HTML Table within TCPDF columns doesnt flow correctly on page break ..." was fixed. 

6.0.089 (2014-07-16)
	- Bug item #948 "bottom line of rowspan cell not work correctly" was fixed.

6.0.088 (2014-07-09)
	- Bug item #946 "Case sensitive type check causes broken match for SVG" was fixed.
	- Bug item #945 "Imagick load doesn't account for passed data string " was fixed.

6.0.087 (2014-06-25)
	- A bug affecting fitcell option in Multicell was fixed.

6.0.086 (2014-06-20)
	- Bug item #938 "Hyphenation-dash extends outside of cell" was fixed (collateral effect).

6.0.085 (2014-06-19)
	- Some example images were replaced.
	- A race condition bug was fixed.
	- Bug item #938 "Hyphenation-dash extends outside of cell" was fixed.

6.0.084 (2014-06-13)
	- A bug related to MultiCell fitcell feature was fixed.
	- Bug item #931 "Documentation error for setPageFormat()" was fixed.

6.0.083 (2014-05-29)
	- Bug item #928 "setHtmlVSpace with HR element" was fixed.

6.0.082 (2014-05-23)
	- Bug item #926 "test statement instead of assignment used in tcpdf_fonts.php" was fixed.
	- Bug item #925 "924 transparent images bug" was fixed.

6.0.081 (2014-05-22)
	- Bug item #922 "writehtml tables thead repeating" was fixed.
	- Patch #71 "External and internal links, local and remote" wa applied.

6.0.080 (2014-05-20)
	- Bug item #921 "Fatal error in hyphenateText() function" was fixed.
	- Bug item #923 "Automatic Hyphenation error" was fixed.
	- Patch #70 "Augument TCPDFBarcode classes with ability to return raw png image data" was applied.

6.0.079 (2014-05-19)
	- Patch item #69 "Named destinations, HTML internal and external links" was merged.
	- Bug item #920 "hyphenateText() should not hyphenate the content of style-tags in HTML mode" was fixed.
	- Image method now trigs an error in case the cache is now writeable.
	- Fixed issue with layer default status.

6.0.078 (2014-05-12)
	- A warning issue in addTTFfont() method was fixed.
	- Fonts were updated to include cbbox metrics.

6.0.077 (2014-05-06)
	- A Datamatrix barcode bug was fixed.

6.0.076 (2014-05-06)
	- A bug in Datamatrix Base256 encoding was fixed.
	- Merged fix for SVG use/clip-gradient.
	- Now it is possible to prefix a page number in Link methods with the * character to avoid been changed when adding/deleting/moving pages (see example_045.php).

6.0.075 (2014-05-05)
	- Bug #917 "Using realtive Units like ex or em for images distort output in HTML mode" was fixed.

6.0.074 (2014-05-03)
	- Part of Bug #917 "Using realtive Units like ex or em for images distort output in HTML mode" was fixed.
	- Bug #915 "Problem with SVG Image using Radial Gradients" was fixed.

6.0.073 (2014-04-29)
	- Bug #913 "Possible bug with line-height" was fixed.
	- Bug #914 "MultiCell and FitCell" was fixed.
	- Bug #915 "Problem with SVG Image using Radial Gradients" was fixed.

6.0.072 (2014-04-27)
	- Deprecated curly braces substring syntax was replaced with square braces.

6.0.071 (2014-04-25)
	- Bug #911 "error with buffered png pics" was fixed.

6.0.070 (2014-04-24)
	- Bug #910 "An SVG image is being cut off (with clipping mask) when you use align options" was fixed.

6.0.069 (2014-04-24)
	- Datamatrix Base256 encoding was fixed.

6.0.068 (2014-04-22)
	- Some Datamatrix barcode bugs were fixed.

6.0.067 (2014-04-21)
	- startLayer() method signature was changed to include a new "lock" parameter.

6.0.066 (2014-04-20)
	- Bug #908 "Linebreak is not considered when getting length of the next string" was fixed.

6.0.065 (2014-04-10)
	- Bug #905 "RGB percentage color bug in convertHTMLColorToDec()" was fixed.

6.0.064 (2014-04-07)
	- Header and Footer fonts are now set by default.
	- Bug #904 "PDF corrupted" was fixed.

6.0.063 (2014-04-03)
	- Method TCPDF_IMAGES::_parsepng() was fixed to support transparency in Indexed images.

6.0.062 (2014-03-02)
	- The method startLayer() now accepts the NULL value for the $print parameter to not set the print layer option.

6.0.061 (2014-02-18)

tcpdf_autoconfig.php
<?php
//============================================================+
// File name   : tcpdf_autoconfig.php
// Version     : 1.0.000
// Begin       : 2013-05-16
// Last Update : 2014-09-02
// Authors     : Nicola Asuni - Tecnick.com LTD - www.tecnick.com - info@tecnick.com
// License     : GNU-LGPL v3 (http://www.gnu.org/copyleft/lesser.html)
// -------------------------------------------------------------------
// Copyright (C) 2011-2014 Nicola Asuni - Tecnick.com LTD
//
// This file is part of TCPDF software library.
//
// TCPDF is free software: you can redistribute it and/or modify it
// under the terms of the GNU Lesser General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// TCPDF is distributed in the hope that it will be useful, but
// WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
// See the GNU Lesser General Public License for more details.
//
// You should have received a copy of the License
// along with TCPDF. If not, see
// <http://www.tecnick.com/pagefiles/tcpdf/LICENSE.TXT>.
//
// See LICENSE.TXT file for more information.
// -------------------------------------------------------------------
//
// Description : Try to automatically configure some TCPDF
//               constants if not defined.
//
//============================================================+

/**
 * @file
 * Try to automatically configure some TCPDF constants if not defined.
 * @package com.tecnick.tcpdf
 * @version 1.0.000
 */

// DOCUMENT_ROOT fix for IIS Webserver
if ((!isset($_SERVER['DOCUMENT_ROOT'])) OR (empty($_SERVER['DOCUMENT_ROOT']))) {
	if(isset($_SERVER['SCRIPT_FILENAME'])) {
		$_SERVER['DOCUMENT_ROOT'] = str_replace( '\\', '/', substr($_SERVER['SCRIPT_FILENAME'], 0, 0-strlen($_SERVER['PHP_SELF'])));
	} elseif(isset($_SERVER['PATH_TRANSLATED'])) {
		$_SERVER['DOCUMENT_ROOT'] = str_replace( '\\', '/', substr(str_replace('\\\\', '\\', $_SERVER['PATH_TRANSLATED']), 0, 0-strlen($_SERVER['PHP_SELF'])));
	} else {
		// define here your DOCUMENT_ROOT path if the previous fails (e.g. '/var/www')
		$_SERVER['DOCUMENT_ROOT'] = '/';
	}
}
$_SERVER['DOCUMENT_ROOT'] = str_replace('//', '/', $_SERVER['DOCUMENT_ROOT']);
if (substr($_SERVER['DOCUMENT_ROOT'], -1) != '/') {
	$_SERVER['DOCUMENT_ROOT'] .= '/';
}

// Load main configuration file only if the K_TCPDF_EXTERNAL_CONFIG constant is set to false.
if (!defined('K_TCPDF_EXTERNAL_CONFIG') OR !K_TCPDF_EXTERNAL_CONFIG) {
	// define a list of default config files in order of priority
	$tcpdf_config_files = array(dirname(__FILE__).'/config/tcpdf_config.php', '/etc/php-tcpdf/tcpdf_config.php', '/etc/tcpdf/tcpdf_config.php', '/etc/tcpdf_config.php');
	foreach ($tcpdf_config_files as $tcpdf_config) {
		if (@file_exists($tcpdf_config) AND is_readable($tcpdf_config)) {
			require_once($tcpdf_config);
			break;
		}
	}
}

if (!defined('K_PATH_MAIN')) {
	define ('K_PATH_MAIN', dirname(__FILE__).'/');
}

if (!defined('K_PATH_FONTS')) {
	define ('K_PATH_FONTS', K_PATH_MAIN.'fonts/');
}

if (!defined('K_PATH_URL')) {
	$k_path_url = K_PATH_MAIN; // default value for console mode
	if (isset($_SERVER['HTTP_HOST']) AND (!empty($_SERVER['HTTP_HOST']))) {
		if(isset($_SERVER['HTTPS']) AND (!empty($_SERVER['HTTPS'])) AND (strtolower($_SERVER['HTTPS']) != 'off')) {
			$k_path_url = 'https://';
		} else {
			$k_path_url = 'http://';
		}
		$k_path_url .= $_SERVER['HTTP_HOST'];
		$k_path_url .= str_replace( '\\', '/', substr(K_PATH_MAIN, (strlen($_SERVER['DOCUMENT_ROOT']) - 1)));
	}
	define ('K_PATH_URL', $k_path_url);
}

if (!defined('K_PATH_IMAGES')) {
	$tcpdf_images_dirs = array(K_PATH_MAIN.'examples/images/', K_PATH_MAIN.'images/', '/usr/share/doc/php-tcpdf/examples/images/', '/usr/share/doc/tcpdf/examples/images/', '/usr/share/doc/php/tcpdf/examples/images/', '/var/www/tcpdf/images/', '/var/www/html/tcpdf/images/', '/usr/local/apache2/htdocs/tcpdf/images/', K_PATH_MAIN);
	foreach ($tcpdf_images_dirs as $tcpdf_images_path) {
		if (@file_exists($tcpdf_images_path)) {
			define ('K_PATH_IMAGES', $tcpdf_images_path);
			break;
		}
	}
}

if (!defined('PDF_HEADER_LOGO')) {
	$tcpdf_header_logo = '';
	if (@file_exists(K_PATH_IMAGES.'tcpdf_logo.jpg')) {
		$tcpdf_header_logo = 'tcpdf_logo.jpg';
	}
	define ('PDF_HEADER_LOGO', $tcpdf_header_logo);
}

if (!defined('PDF_HEADER_LOGO_WIDTH')) {
	if (!empty($tcpdf_header_logo)) {
		define ('PDF_HEADER_LOGO_WIDTH', 30);
	} else {
		define ('PDF_HEADER_LOGO_WIDTH', 0);
	}
}

if (!defined('K_PATH_CACHE')) {
	$K_PATH_CACHE = ini_get('upload_tmp_dir') ? ini_get('upload_tmp_dir') : sys_get_temp_dir();
	if (substr($K_PATH_CACHE, -1) != '/') {
		$K_PATH_CACHE .= '/';
	}
	define ('K_PATH_CACHE', $K_PATH_CACHE);
}

if (!defined('K_BLANK_IMAGE')) {
	define ('K_BLANK_IMAGE', '_blank.png');
}

if (!defined('PDF_PAGE_FORMAT')) {
	define ('PDF_PAGE_FORMAT', 'A4');
}

if (!defined('PDF_PAGE_ORIENTATION')) {
	define ('PDF_PAGE_ORIENTATION', 'P');
}

if (!defined('PDF_CREATOR')) {
	define ('PDF_CREATOR', 'TCPDF');
}

if (!defined('PDF_AUTHOR')) {
	define ('PDF_AUTHOR', 'TCPDF');
}

if (!defined('PDF_HEADER_TITLE')) {
	define ('PDF_HEADER_TITLE', 'TCPDF Example');
}

if (!defined('PDF_HEADER_STRING')) {
	define ('PDF_HEADER_STRING', "by Nicola Asuni - Tecnick.com\nwww.tcpdf.org");
}

if (!defined('PDF_UNIT')) {
	define ('PDF_UNIT', 'mm');
}

if (!defined('PDF_MARGIN_HEADER')) {
	define ('PDF_MARGIN_HEADER', 5);
}

if (!defined('PDF_MARGIN_FOOTER')) {
	define ('PDF_MARGIN_FOOTER', 10);
}

if (!defined('PDF_MARGIN_TOP')) {
	define ('PDF_MARGIN_TOP', 27);
}

if (!defined('PDF_MARGIN_BOTTOM')) {
	define ('PDF_MARGIN_BOTTOM', 25);
}

if (!defined('PDF_MARGIN_LEFT')) {
	define ('PDF_MARGIN_LEFT', 15);
}

if (!defined('PDF_MARGIN_RIGHT')) {
	define ('PDF_MARGIN_RIGHT', 15);
}

if (!defined('PDF_FONT_NAME_MAIN')) {
	define ('PDF_FONT_NAME_MAIN', 'helvetica');
}

if (!defined('PDF_FONT_SIZE_MAIN')) {
	define ('PDF_FONT_SIZE_MAIN', 10);
}

if (!defined('PDF_FONT_NAME_DATA')) {
	define ('PDF_FONT_NAME_DATA', 'helvetica');
}

if (!defined('PDF_FONT_SIZE_DATA')) {
	define ('PDF_FONT_SIZE_DATA', 8);
}

if (!defined('PDF_FONT_MONOSPACED')) {
	define ('PDF_FONT_MONOSPACED', 'courier');
}

if (!defined('PDF_IMAGE_SCALE_RATIO')) {
	define ('PDF_IMAGE_SCALE_RATIO', 1.25);
}

if (!defined('HEAD_MAGNIFICATION')) {
	define('HEAD_MAGNIFICATION', 1.1);
}

if (!defined('K_CELL_HEIGHT_RATIO')) {
	define('K_CELL_HEIGHT_RATIO', 1.25);
}

if (!defined('K_TITLE_MAGNIFICATION')) {
	define('K_TITLE_MAGNIFICATION', 1.3);
}

if (!defined('K_SMALL_RATIO')) {
	define('K_SMALL_RATIO', 2/3);
}

if (!defined('K_THAI_TOPCHARS')) {
	define('K_THAI_TOPCHARS', true);
}

if (!defined('K_TCPDF_CALLS_IN_HTML')) {
	define('K_TCPDF_CALLS_IN_HTML', true);
}

if (!defined('K_TCPDF_THROW_EXCEPTION_ERROR')) {
	define('K_TCPDF_THROW_EXCEPTION_ERROR', false);
}

if (!defined('K_TIMEZONE')) {
	define('K_TIMEZONE', @date_default_timezone_get());
}

//============================================================+
// END OF FILE
//============================================================+
